enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -48,14 +51,18 @@ const InteractiveDashboardInner = ({
     hiddenParameters,
     initialParameterValues,
   });
-
+  const dispatch = useDispatch();
   const { theme } = useEmbedTheme();
 
   const { font } = useEmbedFont();
 
   const store = useStore();
   const [adhocQuestionUrl, setAdhocQuestionUrl] = useState<string | null>(null);
 
+  useUnmount(() => {
```

Comment:
There is one case, we missed is when we drilled a dashboard and then navigated from one dashboard to another dashboard without unmounting e.g. When on a dashboard, click a link that will render another dashboard.
![Screenshot 2024-06-27 at 12 37 56 AM](https://github.com/metabase/metabase/assets/1937582/a8b228ee-9d11-483b-9058-fa09b14aef57)

We need to clear `adhocQuestionUrl` otherwise, we'll keep showing `InteractiveAdHocQuestion` instead of a new dashboard.

I've implemented that with another look
```ts
  useEffect(() => {
    if (dashboardId) {
      dispatch(reset()); // reset "isNavigatingBackToDashboard" state
      setAdhocQuestionUrl(null);
    }
  }, [dashboardId, dispatch]);
```

I also reset `isNavigatingBackToDashboard` here because from a similar scenario.
- Drill a dashboard
- Navigate back (now `isNavigatingBackToDashboard` = `true`)
- Navigate to a different dashboard without unmounting

The `isNavigatingBackToDashboard` would still be `true` if we don't reset it in the hook above.

That was just a solution I came up with. Please feel free to use any solution that suits your situation more.